umaserverdemo/src/main/kotlin/com/lightspark/ReceivingVasp.kt

```diff
@@ -46,15 +60,103 @@ class ReceivingVasp(
     private val nonceCache = InMemoryNonceCache(Clock.System.now().epochSeconds)
     private val coroutineScope = CoroutineScope(Dispatchers.IO)
     private lateinit var senderUmaVersion: String
+    private val httpClient = HttpClient {
+        install(ContentNegotiation) {
+            json(
+                Json {
+                    isLenient = true
+                },
+            )
+        }
+    }
 
     suspend fun createInvoice(call: ApplicationCall): String {
+        val (status, data) = createUmaInvoice(call)
+        if (status != HttpStatusCode.OK) {
+            call.respond(status, data)
+            return data
+        } else {
+            call.respond(data)
+        }
         return "OK"
     }
 
     suspend fun createAndSendInvoice(call: ApplicationCall): String {
+        val senderUma = call.parameters["senderUma"] ?: run {
+            call.respond(HttpStatusCode.BadRequest, "SenderUma not provided.")
+            return "SenderUma not provided."
+        }
+        val (status, data) = createUmaInvoice(call, senderUma)
+        if (status != HttpStatusCode.OK) {
+            call.respond(status, data)
+            return data
+        }
+        val response = try {
+            httpClient.post("/api/uma/payreq/${config.userID}") {
```

Comment:
I'm confused why we're sending to the payreq endpoint. Also what host is this hitting?

Reply:
yeah this one is incorrect I realized when testing - it should be : /api/uma/request_invoice_payment

Reply:
additionally - it should use the domain of the sender, which I have to adjust

Reply:
Shouldn't it be reading the route from the configuration discovery document?

Reply:
done